File: thesis_history/old_code/classes.py
#class

class Node:
    def __init__(self,i_d,edges,interests):
        self.id = i_d
        self.edges = edges
        self.other_interests = interests #if data has attribute for profile interests
        self.neighbors = []
    def set_neighbors(self):
        for edge in self.edges:
            self.neighbors.append(edge.neighhbor_helper(self))
            
class Edge:
    def __init__(self,nodes):
        self.nodes = set(nodes.sort()) #each edge can be indentified by the nodes it connects
    def edge_nodes(self,node): #returns whether or not a node is in an edge
        return self.nodes[0] == node or self.nodes[1] == node
    def neighbor_helper(self,node): #returns the 'other' node in an edge pair
        if self.nodes[0] == node:
            return self.nodes[1]
        return self.nodes[0]
        

File: thesis_history/old_code/classes.cpp
//create node and edge classes

#include <iostream>
#include <vector>
#include <set>
using namespace std;

class Edge;//forward declaration
class Node {
    
  public:
    int id;
    std::vector<Edge> edges;
    std::vector<Node> neighbors;
    void set_neighbors();
    //int area() {return width*height;}
};

class Edge {
    std::set<Node> nodes; //figure out how to pass initalized values here
  public:
    bool edge_nodes (Node node){
        return (this -> nodes.find(node) != this -> nodes.end());
    }
    Node* neighbor_helper(Node node){ //returns the 'other' node in an edge pair
        for (Node element : this -> nodes ){ //iterates through each element of set
            if (element.id != node.id){return *element;}
        }
        return nullptr;
    }

    //int area() {return width*height;}
};

void Node::set_neighbors (){
        for (int i=0; i < this -> edges.size(); i++) {
            this -> neighbors.push_back(edges[i].neighbor_helper(*this));
        }
    }


